pkg/entity-store/postgres.go

```diff
@@ -0,0 +1,268 @@
+package entitystore
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/types"
+	_ "github.com/lib/pq"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+type postgres struct {
+	db *sqlx.DB
+}
+
+type dbEntity struct {
+	Key    string         `db:"key"`
+	Value  types.JSONText `db:"value"`
+	Status string         `db:"status"`
+}
+
+func (p *postgres) createTable() error {
+
+	sql := `
+		CREATE TABLE IF NOT EXISTS entity (
+		key 		TEXT PRIMARY KEY,
```

Comment:
So we are just dumping everything into the JSONB value column?  Because this is SQL, I would at least assume that ID is now the primary key and that name is a field with an index on it.  Based on the discussion earlier, I thought that most/all of the "Base entity" fields would be columns.

Reply:
Probably worth adding a version field as well to deal with future migrations.

Reply:
two sides:
first, we could do that, i.e. put shared attributes/fields into as a DB columns, please take the Status field as an example, which requires manually conversion. We can put other fields as a DB columns as we see necessary.

second, ID can the the primary key, but this change will break the the compatibility with the libkv, and require more work on the manager side, if we do this, two options, refactor and use ID for libkv, or, abandon the libkv implementation.

Reply:
I think this is the implementation maps to libkv, but I if we go all-in with postgres, then it would make sense to add normal columns. JSONB still makes sense for something like `properties` - there is almost always a need to add custom, optional fields per entity (vide labels in k8s, metadata in openstack, etc.) those could be implemented via JSONB. 

However, with our current EntityStore interface (specifically, the List function) I don't think we can take advantage of separate columns (faster filtering and indexes).

Reply:
indeed given the current filter (it is a function) we can do nothing of indexing or filtering with sql, I consider to refactor the filter to a map-liked structure, so that we could create dynamic sql queries.

for example:
```
filter := map[string]string{
"status":"ready"
}
```
generate a sql like this 
```
select * from entity where status = 'ready'
```
along with an index on columns or within a JSONB field, (for example https://www.compose.com/articles/faster-operations-with-the-jsonb-data-type-in-postgresql/)

I am optimistic to that. especially most of our use case that requires a filter are equal-based comparison .  

PS. JSONB field will also be used for the custom fields of a specific entity (e.g. API, Image, ...) that the BaseEntity don't contain.